File: app/src/main/java/com/github/satoshun/example/dagger/android/contribute/main/MainFragment.java
package com.github.satoshun.example.dagger.android.contribute.main;

import android.os.Bundle;
import dagger.android.support.DaggerFragment;

import javax.inject.Inject;

public class MainFragment extends DaggerFragment {

  @Inject String message; // from App Component
  @Inject int tax; // from Activity Component
  @Inject float weight; // from Fragment Component

  @Override public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setRetainInstance(true);
  }
}

File: app/src/main/java/com/github/satoshun/example/dagger/android/contribute/main/MainFragment.kt
package com.github.satoshun.example.dagger.android.contribute.main

import android.content.Context
import androidx.fragment.app.Fragment
import com.github.satoshun.example.dagger.android.contribute.Hoge
import dagger.android.support.AndroidSupportInjection
import javax.inject.Inject

class MainFragment : Fragment() {
  @Inject lateinit var hoge: Hoge

  override fun onAttach(context: Context) {
    AndroidSupportInjection.inject(this)
    super.onAttach(context)

    hoge.show()
  }
}
